### DISCUSSION TITLE
Peripheral Register View Plugin

### DISCUSSION THREAD

**Discussion by avi-77** [eclipse-embed-cdt/eclipse-plugins#598] [Debug Plug-ins Q&A]
Hi @ilg-ul,

The peripheral register view plugin displays an issue where the value updates properly, but the corresponding description remains static or fails to update accordingly.

**Steps to Reproduce:**

1. Open the peripheral register view plugin.
2. Select any peripheral or register.
3. Modify the value of the selected register.
4. Observe that while the value changes correctly, the description associated with the register fails to update or remains unchanged.

**Expected Behavior:**
When a register's value is modified, the associated description should also update to reflect the changes made, providing accurate information about the register's state.

**Actual Behavior:**
The register value updates correctly, but the corresponding description does not reflect the changes made, remaining static or outdated.

**Environment:**
Plugin Version: 6.3.2
Operating System: Windows 10
Development Environment: Eclipse 2023-03

**Additional Notes:**
This issue affects the accuracy of information displayed to developers using the peripheral register view plugin. It may lead to confusion and incorrect interpretations of the register values due to mismatched or outdated descriptions.

**Attachments:**
I've included screenshots displaying the issue where the description value remains unchanged until the user clicks on the specific field.
![outdated_description](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/79654418/c5907ec2-3b0b-4637-a75d-d27584bcc8bb)
![updated_description](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/79654418/ae574390-ad31-401f-87a9-f4ef9edbffbf)

Regards,

Avi

**Comment by ilg-ul** [comment]
Thank you for the report. I'm not sure I understand it properly.

What device was it, and what SVD file?

**Reply by avi-77** [reply]
Hi @ilg-ul: The device in question is the RSL15, and I've included the svd file in text format. The problem lies in the GPIO_OUTPUT_DATA->DATA value, which ideally should be **0x100: GPIO8_OUT_HIGH**. However, it's currently displaying **0x100: GPIO0_OUT_LOW** until the user clicks on this field. The numeric value updates by default, but the description remains outdated.

[rsl15.txt](https://github.com/eclipse-embed-cdt/eclipse-plugins/files/13416090/rsl15.txt)
